File: 프로그래머스/0/181835. 조건에 맞게 수열 변환하기 3/조건에 맞게 수열 변환하기 3.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int k) {
        int[] answer = arr;
        boolean odd = k % 2 != 0 ? true : false;
        
        for(int i = 0; i < arr.length; i++) answer[i] = odd ? arr[i] * k : arr[i] + k;
        
        return answer;
    }
}

File: 프로그래머스/0/181835. 조건에 맞게 수열 변환하기 3/조건에 맞게 수열 변환하기 3.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len, int k) {

    int* answer = (int*)malloc(arr_len * sizeof(int));
    
    int check = k % 2 == 0 ? 1 : 0;
    
    for(int i = 0; i < arr_len; i++){
        if(check){
            answer[i] = arr[i] + k;
        }else {
            answer[i] = arr[i] * k;
        }
    }
    
    return answer;
}